### ISSUE TITLE
when restarting, stream will not always wait for the stream to end before starting up

### ISSUE THREAD

**Issue by brandonhs** [DeepwaterExploration/DWE_OS#395]

**Comment by brandonhs** [comment]
Fixed by adding Popen.wait() after killing the subprocess.

Another issue fixed with the same effect by limiting the number of streams that can be added to a leader to avoid duplicates.
